### PR TITLE
Dev/177 Compilation script API integrate

### PR DESCRIPTION
**Pull request by Chaoyuuu**
Solved #177 

image

### CODE REVIEW

**Review by Johnny850807 — APPROVED** [review]
LGTM

File: src/app/problem/Model.js [hunk 1]
```diff
@@ -8,6 +8,14 @@ export class LanguageEnv {
         this.submittedCodeSpecs = submittedCodeSpecs;
     }
 
+    getCompilationScript() {
+        return this.compilationScript
+    }
+
+    updateCompilationScript(script) {
+        this.compilationScript = script
```
Comment (Wally5077) [inline src/app/problem/Model.js:16]: ```;```

File: src/app/problem/editor/right/CompilationScript.js [hunk 2]
```diff
@@ -1,59 +1,84 @@
...
     const onClickEdit = () => {
         setIsEditing(true);
-    }
+        setScriptBackUp(script);
+    };
 
     const onClickSave = () => {
         setIsEditing(false);
-    }
+        languageEnv.updateCompilationScript(script)
```
Comment (Wally5077) [inline src/app/problem/editor/right/CompilationScript.js:39]: ```;```

File: src/app/problem/editor/right/CompilationScript.js [hunk 3]
```diff
@@ -1,59 +1,84 @@
...
+        problemService.updateLanguageEnv(problemId, languageEnv)
+            .then(() => {
+                console.log("The problem's compilation script has been updated");
+            });
+    };
 
     const onClickCancel = () => {
         setIsEditing(false);
-    }
+        setScript(scriptBackUp)
```
Comment (Wally5077) [inline src/app/problem/editor/right/CompilationScript.js:48]: ```;```

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

**Review by Wally5077 — APPROVED** [review]
LGTM

**Pull request merged**